Support Pydantic validation context (#3448)
pydantic/pydantic-ai: docs/tools-advanced.md

## Tool Execution and Retries {#tool-retries}

> When a tool is executed, its arguments (provided by the LLM) are first validated against the function's signature using Pydantic. Note that the [validation context](output.md#validation-context) - if provided - will be used during this validation. If validation fails (e.g., due to incorrect types or missing required arguments), a `ValidationError` is raised, and the framework automatically generates a [`RetryPromptPart`][pydantic_ai.messages.RetryPromptPart] containing the validation details. This prompt is sent back to the LLM, informing it of the error and allowing it to correct the parameters and retry the tool call.

Reviewer: ```suggestion
When a tool is executed, its arguments (provided by the LLM) are first validated against the function's signature using Pydantic (with optional [validation context](output.md#validation-context)). If validation fails (e.g., due to incorrect types or missing required arguments), a `ValidationError` is raised, and the framework automatically generates a [`RetryPromptPart`][pydantic_ai.messages.RetryPromptPart] containing the validation details. This prompt is sent back to the LLM, informing it of the error and allowing it to correct the parameters and retry the tool call.
```
